beacon-chain/core/epoch/precompute/new.go

```diff
@@ -17,9 +16,13 @@ import (
 // New gets called at the beginning of process epoch cycle to return
 // pre computed instances of validators attesting records and total
 // balances attested in an epoch.
-func New(ctx context.Context, state *stateTrie.BeaconState) ([]*Validator, *Balance, error) {
+func New(ctx context.Context, state iface.BeaconState) ([]*Validator, *Balance, error) {
 	ctx, span := trace.StartSpan(ctx, "precomputeEpoch.New")
 	defer span.End()
+	if state == nil || state.InnerStateUnsafe() == nil {
```

Comment:
same as what we discussed earlier, the interface cannot be nil. So the check will not check for anything.